src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py

```diff
@@ -205,9 +210,15 @@ def normalize_disk_info(image_data_disks=None,
 
         if is_valid_resource_id(d):
             info[i]['managedDisk'] = {'id': d}
+            if data_disk_delete_option:
+                info[i]['deleteOption'] = data_disk_delete_option if isinstance(data_disk_delete_option, str) \
```

Comment:
If data_disk_delete_option can't be string type, could we remove these judgments?
> if isinstance(data_disk_delete_option, str)

Reply:
this data_disk_delete_option could be str after validation here:
https://github.com/Azure/azure-cli/blob/69dfa2292386ee5c82babf95945b694858890a76/src/azure-cli/azure/cli/command_modules/vm/_validators.py#L921-L923

Reply:
OK~